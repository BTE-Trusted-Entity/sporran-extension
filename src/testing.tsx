export * from '@testing-library/react';
export { waitForElementToBeRemoved } from '@testing-library/dom';
import { act, render as externalRender, screen } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import dialogPolyfill from 'dialog-polyfill';

import { AccountsProviderMock } from './testing/AccountsProviderMock';

export {
  AccountsProviderMock,
  accountsMock,
} from './testing/AccountsProviderMock';

export { mockBackgroundScript } from './testing/mockBackgroundScript';

export function render(
  ui: Parameters<typeof externalRender>[0],
  options?: Parameters<typeof externalRender>[1],
): ReturnType<typeof externalRender> {
  return externalRender(
    <AccountsProviderMock>
      <MemoryRouter>{ui}</MemoryRouter>
    </AccountsProviderMock>,
    options,
  );
}

export async function waitForTooltipUpdate(): Promise<void> {
  expect(await screen.findByRole('status')).toHaveAttribute(
    'data-popper-escaped',
  );
}

const dialogPromise = Promise.resolve();
(dialogPolyfill.registerDialog as jest.Mock).mockReturnValue(dialogPromise);

export async function waitForDialogUpdate(): Promise<void> {
  await act(() => dialogPromise);
}
