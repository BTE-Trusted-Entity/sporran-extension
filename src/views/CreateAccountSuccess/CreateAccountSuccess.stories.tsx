import { Meta } from '@storybook/react';

import { CreateAccountSuccess } from './CreateAccountSuccess';

export default {
  title: 'Views/CreateAccountSuccess',
  component: CreateAccountSuccess,
} as Meta;

export function Create(): JSX.Element {
  return <CreateAccountSuccess />;
}

export function Import(): JSX.Element {
  return <CreateAccountSuccess type="import" />;
}
