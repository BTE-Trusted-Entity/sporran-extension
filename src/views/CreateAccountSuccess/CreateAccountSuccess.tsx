import { browser } from 'webextension-polyfill-ts';
import { Link } from 'react-router-dom';

interface Props {
  type?: 'create' | 'import';
}

export function CreateAccountSuccess({ type = 'create' }: Props): JSX.Element {
  const t = browser.i18n.getMessage;
  return (
    <main>
      <h1>{t('view_CreateAccountSuccess_heading')}</h1>

      <p>
        {type === 'create' && t('view_CreateAccountSuccess_message_create')}
        {type === 'import' && t('view_CreateAccountSuccess_message_import')}
      </p>

      <Link to="/">{t('view_CreateAccountSuccess_CTA')}</Link>
    </main>
  );
}
