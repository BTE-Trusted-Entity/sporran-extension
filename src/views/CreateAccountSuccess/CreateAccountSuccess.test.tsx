import { CreateAccountSuccess } from './CreateAccountSuccess';
import { render } from '../../testing';

describe('CreateAccountSuccess', () => {
  it('should render', async () => {
    const { container } = render(<CreateAccountSuccess />);
    expect(container).toMatchSnapshot();
  });
});
