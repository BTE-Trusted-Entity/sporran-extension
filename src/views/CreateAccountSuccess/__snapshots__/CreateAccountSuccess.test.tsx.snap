// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreateAccountSuccess should render 1`] = `
<div>
  <main>
    <h1>
      Congratulations
    </h1>
    <p>
      You have successfully created a KILT-Wallet
    </p>
  </main>
</div>
`;
